cache/cache.go

```diff
@@ -0,0 +1,31 @@
+package cache
+
+import (
+	"opg-file-service/session"
+	"os"
+
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-secretsmanager-caching-go/secretcache"
+)
+
+type SecretsCache struct {
+	env   string
+	cache *secretcache.Cache
+}
+
+func applyAwsConfig(c *secretcache.Cache) {
+	session, _ := session.NewSession()
+	endpoint := os.Getenv("SECRETS_MANAGER_ENDPOINT")
+	session.AwsSession.Config.Endpoint = &endpoint
+	c.Client = secretsmanager.New(session.AwsSession)
+}
+
+func New() *SecretsCache {
```

Comment:
Still early in my Go journey but this function, what is it exactly doing?

Reply:
Go doesn't really have constructors but if you want to create a struct that has private (lowercase) fields outside of the package, you need to create a factory function that returns it. The difference is constructors in OOP are methods of a class, whereas Golang factory functions are functions of a package.

The convention (as far as I am aware) is to name these functions `New` if they are the only factory function in the package, or `New<StructName>` if there are multiple.